src/tpool.c

```diff
@@ -114,11 +116,42 @@ int tpool_add_work(tpool_t *pool, thread_func_t func, void *arg) {
     return 1;
 }
 
+/**
+ * see: https://github.com/htop-dev/htop/blob/f782f821f7f8081cb43bbad1c37f32830a260a81/linux/LinuxProcessList.c
+ */
+__always_inline
+static size_t _get_total_mem() {
```

Comment:
I guess it doesn't really matter since it wouldn't make much sense to cache the results anyway..

Reply:
I tried to deliberately set the threshold too low so 16 threads poll memory consumption with `usleep(10000)`. CPU utilization is around 1%~3% with some kernel time. I think this is acceptable? Consider it would also need milliseconds to process a file.

Reply:
Yes this is perfect, thanks